### PR TITLE
Set unified_mode for custom resources

### PR DESCRIPTION
**Pull request by detjensrobert**
Also cleans up some unneeded files to match our other cookbooks.

### CODE REVIEW

**Comment by detjensrobert** [comment]
[Chef 17: Enable unified_mode on all custom resources](https://trello.com/c/1AanV573/2378-chef-17-enable-unifiedmode-on-all-custom-resources)

**Comment by ramereth** [comment]
So ideally I'd like to be able to run Chef 17 on here with this change.

1. Let's go ahead and remove the `lvm` cookbook and just replace it with a `package` resource which installs `lvm2`.
2. I'd like to remove the `hostsfile` cookbook. We can probably do what we need with the `line` cookbook FWIW by just appending what we need to the file.

**Comment by detjensrobert** [comment]
Should this cookbook use `osl-repos` instead of the yum cookbooks?

File: kitchen.yml [hunk 1]
```diff
@@ -48,6 +40,8 @@ suites:
       ganeti:
         master-node: true
         drbd: true
+    provisioner:
+      multiple_converge: 3 # to fix directory ownership contention
```
Comment (ramereth) [inline kitchen.yml:44]: Looks like this changed with the release of Ganeti 3.x and probably should be fixed in the recipe.
```
         * directory[/var/lib/ganeti/rapi] action create
           - change owner from 'gnt-rapi' to 'root'
           - change group from 'gnt-masterd' to 'root'
           - restore selinux security context
```

File: recipes/default.rb [hunk 2]
```diff
@@ -46,15 +47,13 @@
...
   mode '700'
   recursive true
-  action :create
 end
 
 directory '/var/lib/ganeti/rapi' do
-  owner 'root'
-  group 'root'
+  owner 'gnt-rapi'
+  group 'gnt-masterd'
```
Comment (ramereth) [inline recipes/default.rb:54]: So, as I said, this only need to happen on Ganeti >= 3.0.0. We should make a helper method that uses the info from the following to get the version of the installed version:
```
[root@gprod1.osuosl.bak ~]# gnt-cluster version
Software version: 2.16.1
Internode protocol: 2160000
Configuration format: 2160000
OS api version: 20
Export interface: 0
VCS version: (ganeti) version v2.16.1
```

Comment (detjensrobert) [reply recipes/default.rb:54]: Should this use the `node['ganeti']['version']` attribute to pull the version instead of shelling out?

Comment (ramereth) [reply recipes/default.rb:54]: @detjensrobert hmm, that would work but we'd need to update osl-ganeti to set that (as we use a local attribute currently). We can assume `nil` is `>= 3.0`

**Review by ramereth — APPROVED** [review]
Don't merge this until we deal with the version issue properly in osl-ganeti

Also, please do a major bump.

**Comment by ramereth** [comment]
!bump major ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a major-level version bump to v5.0.0.  Have a nice day!

**Pull request merged**